### PR TITLE
first pass at env config refactor

### PR DESCRIPTION
**Pull request by ekemper**

### CODE REVIEW

File: cstar/base/environment.py [hunk 1]
```diff
@@ -25,170 +23,19 @@
...
-    # TODO: lots of this is repeat code, can determine a lot of these vars using functions rather than hardcoding
+# TODO: what is a cleaner way to destructure this as one would in JS?
+_CSTAR_COMPILER = runtimeConfig.envVars._CSTAR_COMPILER
+_CSTAR_SYSTEM = runtimeConfig.envVars._CSTAR_SYSTEM
+_CSTAR_SCHEDULER = runtimeConfig.envVars._CSTAR_SCHEDULER
+_CSTAR_ENVIRONMENT_VARIABLES = runtimeConfig.envVars._CSTAR_ENVIRONMENT_VARIABLES
+_CSTAR_SYSTEM_DEFAULT_PARTITION = runtimeConfig.envVars._CSTAR_SYSTEM_DEFAULT_PARTITION
+_CSTAR_SYSTEM_CORES_PER_NODE = runtimeConfig.envVars._CSTAR_SYSTEM_CORES_PER_NODE
+_CSTAR_SYSTEM_MEMGB_PER_NODE = runtimeConfig.envVars._CSTAR_SYSTEM_MEMGB_PER_NODE
+_CSTAR_SYSTEM_MAX_WALLTIME = runtimeConfig.envVars._CSTAR_SYSTEM_MAX_WALLTIME
```
Comment (ekemper) [inline cstar/base/environment.py]: pulling the env vars in from the config class

File: cstar/base/hpc_env_var_map.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import os
```
Comment (ekemper) [inline cstar/base/hpc_env_var_map.py:1]: This file is responsible for just the env vars that are meaningful in the the HPC envs

File: cstar/base/runtime_env_config.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        self._CSTAR_SYSTEM_MEMGB_PER_NODE = ''
+        self._CSTAR_SYSTEM_MAX_WALLTIME = ''
+        
+        if (self.system == "Linux") and ("LMOD_DIR" in list(os.environ)):
+            self.configureHPCRuntimeEnv()
+        else:
+            self.configureLocalEnv() 
+    
+    
+    def loadEnvModule():
```
Comment (ekemper) [inline cstar/base/runtime_env_config.py]: `LoadEnvModule()` and `loadLinuxEnvModule()` are broken out from the original if statement in `environment.py`

Perhaps the names could be revised to be more contextually meaningful.

**Comment by dafyddstephenson** [comment]
Hi Ed, thanks for your interest in this! There is an open PR (#168) that achieves some of the things you have begun working on, that you may wish to take a look at or contribute to :)

**Comment by ekemper** [comment]
@dafyddstephenson Thanks for the heads up! I accidentally created the PR on the original repo instead of my fork . OOPs!
Ill checkout the existing PR.

**Pull request closed without merging**